src/StarkEx.Crypto.SDK/Signing/StarkExSigner.cs

```diff
@@ -91,4 +133,62 @@ private static BigInteger FixMessageLength(BigInteger messageHash)
             _ => throw new ArgumentException($"Invalid message hash with length {hashHex.Length}", nameof(messageHash)),
         };
     }
+
+    private static int CalculateByteLength(int length, int byteSize = 8)
+    {
+        var remainder = length % byteSize;
+        return remainder > 0 ?
+            (((length - remainder) / byteSize) * byteSize) + byteSize :
+            length;
+    }
+
+    private static string GrindKey(string keySeed, BigInteger keyValLimit)
+    {
+        var sha256EcMaxDigest = new BigInteger("2").Pow(256);
+        var maxAllowedVal = sha256EcMaxDigest.Subtract(sha256EcMaxDigest.Mod(keyValLimit));
+
+        var i = 0;
+        var key = HashKeyWithIndex(keySeed, i++);
```

Comment:
nitpick: I know that the `i` var is only incremented after the method call, but this can sometimes case some confusion.

wdyt of moving the `i++` to a new line to improve readability?

Reply:
I actually find it more readable to have the original way but given that both of you prefer to have the increment on a separate line, I've changed it. To be fair it is also the way StarkWare implemented this code block